client/trino-client/src/main/java/io/trino/client/StatementClientV1.java

```diff
@@ -100,7 +101,10 @@ public StatementClientV1(OkHttpClient httpClient, ClientSession session, String
         this.timeZone = session.getTimeZone();
         this.query = query;
         this.requestTimeoutNanos = session.getClientRequestTimeout();
-        this.user = session.getUser().orElse(session.getPrincipal());
+        this.user = Stream.of(session.getUser(), session.getPrincipal())
+                .filter(Optional::isPresent)
```

Comment:
This simplifies to
```java
.flatMap(Optional::stream)

Reply:
`Optional.stream` was added in Java 9 so it can't be used in `trino-client` which targets Java 8.